### PR TITLE
Pawn promotion

### PR DESCRIPTION
**Pull request by nocommentcode**
Implemented the Pawn Promotion Functionality
When the user drags a Pawn into the final row on either side, they will be promted to select their desired promotion
Demo:

https://user-images.githubusercontent.com/26475724/170444448-6e0c5a8d-27f9-453e-83c5-e0255a54b066.mp4

### CODE REVIEW

File: packages/TorneloScoresheet/__tests__/chessEngine.test.ts [hunk 1]
```diff
@@ -949,3 +949,67 @@ describe('makeMove', () => {
...
+      san: 'cxb8',
+      move: { from: 'c7', to: 'b8' },
+      promotion: true,
+    },
+  ];
+
+  testCases.forEach(function (testCase) {
+    it(testCase.name, function () {
+      expect(
+        chessEngine.isPawnPromotion(testCase.fen, testCase.move as PlySquares),
```
Comment (gestureleft) [inline packages/TorneloScoresheet/__tests__/chessEngine.test.ts:1011]: Should this be `as MoveSquares`?

**Review by gestureleft — CHANGES_REQUESTED** [review]
Looks good! It isn't compiling (see the comment I've left), but other than that looks great!

File: packages/TorneloScoresheet/__tests__/chessEngine.test.ts (hunk 1 above)
Comment (nocommentcode) [reply packages/TorneloScoresheet/__tests__/chessEngine.test.ts]: Yeh weird as the test was still running and passing? Not sure how that's possible

**Review by gestureleft — APPROVED** [review]
LGTM

**Pull request merged**